File: app/src/main/java/com/nps/sample/MainActivity.java
package com.nps.sample;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.nps.adiscope.AdiscopeError;
import com.nps.adiscope.AdiscopeSdk;
import com.nps.adiscope.interstitial.InterstitialAd;
import com.nps.adiscope.interstitial.InterstitialAdListener;
import com.nps.adiscope.listener.AdiscopeInitializeListener;
import com.nps.adiscope.offerwall.OfferwallAd;
import com.nps.adiscope.offerwall.OfferwallAdListener;
import com.nps.adiscope.reward.RewardItem;
import com.nps.adiscope.reward.RewardedVideoAd;
import com.nps.adiscope.reward.RewardedVideoAdListener;

/**
 * readme
 *   - admob rewarded video network
 *     - need to add meta-data (com.google.android.gms.ads.APPLICATION_ID) to AndroidManifest.xml
 *
 *   - fan rewarded video network
 *     - need to register test device id to facebook admin
 *
 */
public class MainActivity extends Activity implements RewardedVideoAdListener, OfferwallAdListener, InterstitialAdListener {

    private static final String TAG = MainActivity.class.getName();

    private String SAMPLE_MEDIA_ID              = "";
    private String SAMPLE_MEDIA_SECRET          = "";
    private String SAMPLE_OFFERWALL_UNIT        = "";
    private String SAMPLE_REWARDED_VIDEO_UNIT   = "";
    private String SAMPLE_INTERSTITIAL_UNIT     = "";

    private static OfferwallAd mOfferwallAd;
    private static RewardedVideoAd mRewardedVideoAd;
    private static InterstitialAd mInterstitialAd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // initialize adiscope sdk
        AdiscopeSdk.initialize(this, SAMPLE_MEDIA_ID, SAMPLE_MEDIA_SECRET, new AdiscopeInitializeListener() {
            @Override
            public void onInitialized(boolean isSuccess) {
                if (isSuccess) {
                    // get offerwall singleton instance
                    mOfferwallAd = AdiscopeSdk.getOfferwallAdInstance(MainActivity.this);

                    // get rewardVideo singleton instance
                    mRewardedVideoAd = AdiscopeSdk.getRewardedVideoAdInstance(MainActivity.this);

                    // get interstitial singleton instance
                    mInterstitialAd = AdiscopeSdk.getInterstitialAdInstance(MainActivity.this);

                    // set listener
                    mOfferwallAd.setOfferwallAdListener(MainActivity.this);
                    mRewardedVideoAd.setRewardedVideoAdListener(MainActivity.this);
                    mInterstitialAd.setInterstitialAdListener(MainActivity.this);
                }
            }
        });

        // set userId (user unique id)
        AdiscopeSdk.setUserId("testUserId");

        ((Button)findViewById(R.id.btnShowOfferwall)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // show offerwall

                String[] excludeType = new String[]{}; // excludeType List
                // ex) [CPI, CPE, CPS, CPA]
                mOfferwallAd.show(MainActivity.this, SAMPLE_OFFERWALL_UNIT, excludeType);
            }
        });

        ((Button)findViewById(R.id.btnLoadVideo)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // load rewarded video
                mRewardedVideoAd.load(SAMPLE_REWARDED_VIDEO_UNIT);
            }
        });

        ((Button)findViewById(R.id.btnLoadInterstitial)).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // load interstitial
                mInterstitialAd.load(SAMPLE_INTERSTITIAL_UNIT);
            }
        });

    }

    // region implementation RewardedVideoAdListener
    @Override
    public void onRewardedVideoAdLoaded(String unitId) {
        Log.d(TAG, "onRewardedVideoAdLoaded");
        mRewardedVideoAd.show();
    }

    @Override
    public void onRewardedVideoAdFailedToLoad(String unitId, AdiscopeError adiscopeError) {
        Log.e(TAG, "onRewardedVideoAdFailedToLoad : " + adiscopeError);
    }

    @Override
    public void onRewardedVideoAdOpened(String unitId) {
        Log.d(TAG, "onRewardedVideoAdOpened");

    }

    @Override
    public void onRewardedVideoAdClosed(String unitId) {
        Log.d(TAG, "onRewardedVideoAdClosed");

    }

    @Override
    public void onRewarded(String unitId, RewardItem rewardItem) {
        Log.d(TAG, "onRewarded");

    }

    @Override
    public void onRewardedVideoAdFailedToShow(String unitId, AdiscopeError adiscopeError) {
        Log.e(TAG, "onRewardedVideoAdFailedToLoad : " + adiscopeError);

    }
    // endregion

    // region implementation OfferwallAdListener
    @Override
    public void onOfferwallAdOpened(String unitId) {
        Log.d(TAG, "onOfferwallAdOpened");

    }

    @Override
    public void onOfferwallAdFailedToShow(String unitId, AdiscopeError adiscopeError) {
        Log.e(TAG, "onRewardedVideoAdFailedToLoad : " + adiscopeError);
    }

    @Override
    public void onOfferwallAdClosed(String unitId) {
        Log.d(TAG, "onOfferwallAdClosed");

    }
    // endregion

    // region implementation InterstitialAdListener
    @Override
    public void onInterstitialAdLoaded() {
        Log.d(TAG, "onInterstitialAdLoaded");
        mInterstitialAd.show();
    }

    @Override
    public void onInterstitialAdFailedToLoad(AdiscopeError adiscopeError) {
        Log.e(TAG, "onInterstitialAdFailedToLoad : " + adiscopeError);
    }

    @Override
    public void onInterstitialAdOpened(String unitId) {
        Log.d(TAG, "onInterstitialAdOpened");
    }

    @Override
    public void onInterstitialAdClosed(String unitId) {
        Log.d(TAG, "onInterstitialAdClosed");
    }

    @Override
    public void onInterstitialAdFailedToShow(String unitId, AdiscopeError adiscopeError) {
        Log.e(TAG, "onInterstitialAdFailedToShow : " + adiscopeError);
    }
    // endregion
}

File: app/src/main/java/com/nps/sample/MainActivity.kt
package com.nps.sample

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import com.nps.adiscope.AdiscopeError
import com.nps.adiscope.AdiscopeSdk
import com.nps.adiscope.interstitial.InterstitialAd
import com.nps.adiscope.interstitial.InterstitialAdListener
import com.nps.adiscope.offerwall.OfferwallAd
import com.nps.adiscope.offerwall.OfferwallAdListener
import com.nps.adiscope.reward.RewardItem
import com.nps.adiscope.reward.RewardedVideoAd
import com.nps.adiscope.reward.RewardedVideoAdListener

class MainActivity : Activity(), RewardedVideoAdListener, OfferwallAdListener, InterstitialAdListener {
    private val TAG = MainActivity::class.java.name

    private val SAMPLE_MEDIA_ID = "215"
    private val SAMPLE_MEDIA_SECRET = "77d1c135490f42adb89f0ed0e14f00b2"
    private val SAMPLE_OFFERWALL_UNIT = ""
    private val SAMPLE_REWARDED_VIDEO_UNIT = "MC_RV"
    private val SAMPLE_INTERSTITIAL_UNIT = ""

    private var mOfferwallAd: OfferwallAd? = null
    private var mRewardedVideoAd: RewardedVideoAd? = null
    private var mInterstitialAd: InterstitialAd? = null

    private var offerwallBtn: Button? = null
    private var rvBtn: Button? = null
    private var interstitialBtn: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        offerwallBtn = findViewById(R.id.btnShowOfferwall)
        rvBtn = findViewById(R.id.btnLoadVideo)
        interstitialBtn = findViewById(R.id.btnLoadInterstitial)

        // initialize adiscope sdk
        AdiscopeSdk.initialize(
            this@MainActivity, SAMPLE_MEDIA_ID, SAMPLE_MEDIA_SECRET
        ) { isSuccess ->
            if (isSuccess) {
                // get offerwall singleton instance
                mOfferwallAd = AdiscopeSdk.getOfferwallAdInstance(this@MainActivity)

                // get rewardVideo singleton instance
                mRewardedVideoAd = AdiscopeSdk.getRewardedVideoAdInstance(this@MainActivity)

                // get interstitial singleton instance
                mInterstitialAd = AdiscopeSdk.getInterstitialAdInstance(this@MainActivity)

                // set listener
                mOfferwallAd?.setOfferwallAdListener(this@MainActivity)
                mRewardedVideoAd?.setRewardedVideoAdListener(this@MainActivity)
                mInterstitialAd?.setInterstitialAdListener(this@MainActivity)
                offerwallBtn?.isEnabled = true
                rvBtn?.isEnabled = true
                interstitialBtn?.isEnabled = true
            } else {
                // Init 실패 에 대한 처리 Code
            }
        }

        // set userId (user unique id)
        AdiscopeSdk.setUserId("testUserId")
        offerwallBtn?.setOnClickListener(View.OnClickListener { // show offerwall
            val excludeType = arrayOf<String>() // excludeType List
            // ex) [CPI, CPE, CPS, CPA]
            mOfferwallAd?.show(this@MainActivity, SAMPLE_OFFERWALL_UNIT, excludeType)
        })
        rvBtn?.setOnClickListener(View.OnClickListener { // load rewarded video
            mRewardedVideoAd?.load(SAMPLE_REWARDED_VIDEO_UNIT)
        })
        interstitialBtn?.setOnClickListener(View.OnClickListener { // load interstitial
            mInterstitialAd?.load(SAMPLE_INTERSTITIAL_UNIT)
        })
    }

    // region implementation RewardedVideoAdListener
    override fun onRewardedVideoAdLoaded(unitId: String?) {
        Log.d(TAG, "onRewardedVideoAdLoaded")
        mRewardedVideoAd!!.show()
    }

    override fun onRewardedVideoAdFailedToLoad(unitId: String?, adiscopeError: AdiscopeError) {
        Log.e(TAG, "onRewardedVideoAdFailedToLoad : $adiscopeError")
    }

    override fun onRewardedVideoAdOpened(unitId: String?) {
        Log.d(TAG, "onRewardedVideoAdOpened")
    }

    override fun onRewardedVideoAdClosed(unitId: String?) {
        Log.d(TAG, "onRewardedVideoAdClosed")
    }

    override fun onRewarded(unitId: String?, rewardItem: RewardItem?) {
        Log.d(TAG, "onRewarded")
    }

    override fun onRewardedVideoAdFailedToShow(unitId: String?, adiscopeError: AdiscopeError) {
        Log.e(TAG, "onRewardedVideoAdFailedToLoad : $adiscopeError")
    }
    // endregion

    // endregion
    // region implementation OfferwallAdListener
    override fun onOfferwallAdOpened(unitId: String?) {
        Log.d(TAG, "onOfferwallAdOpened")
    }

    override fun onOfferwallAdFailedToShow(unitId: String?, adiscopeError: AdiscopeError) {
        Log.e(TAG, "onRewardedVideoAdFailedToLoad : $adiscopeError")
    }

    override fun onOfferwallAdClosed(unitId: String?) {
        Log.d(TAG, "onOfferwallAdClosed")
    }
    // endregion

    // endregion
    // region implementation InterstitialAdListener
    override fun onInterstitialAdLoaded() {
        Log.d(TAG, "onInterstitialAdLoaded")
        mInterstitialAd!!.show()
    }

    override fun onInterstitialAdFailedToLoad(adiscopeError: AdiscopeError) {
        Log.e(TAG, "onInterstitialAdFailedToLoad : $adiscopeError")
    }

    override fun onInterstitialAdOpened(unitId: String?) {
        Log.d(TAG, "onInterstitialAdOpened")
    }

    override fun onInterstitialAdClosed(unitId: String?) {
        Log.d(TAG, "onInterstitialAdClosed")
    }

    override fun onInterstitialAdFailedToShow(unitId: String?, adiscopeError: AdiscopeError) {
        Log.e(TAG, "onInterstitialAdFailedToShow : $adiscopeError")
    }
}